tracer.go

```diff
@@ -23,4 +21,22 @@ type Tracer interface {
 	//         SetTag("lucky_number", 42)
 	//
 	StartTrace(operationName string) Span
+
+	// Return a PropagationExtractor for the given `format` value, or nil if the
+	// Span does not support such a format.
+	//
+	// OpenTracing defines a common set of `format` values (see
+	// BuiltinPropagationFormat), and each has an expected carrier type.
+	//
+	// Other packages may declare their own `format` values, much like the keys
```

Comment:
(FWIW, I would not imagine this pattern working well in non-Go... other languages could just use `String` for their `format` params)